src/main/java/nextstep/courses/service/StudentService.java

```diff
@@ -32,4 +36,14 @@ public Student findById(Long id) {
     public List<Student> findBySessionId(Long sessionId) {
         return studentRepository.findBySessionId(sessionId);
     }
+  
+    @Transactional
+    public int approveStudent(Long sessionId, Long id) {
+        return studentRepository.updateApprovalState(sessionId, id, ApprovalState.APPROVAL);
+    }
+
+    @Transactional
+    public int approveCancelStudent(Long sessionId, Long id) {
+        return studentRepository.updateApprovalState(sessionId, id, ApprovalState.UN_APPROVAL);
+    }
```

Comment:
```suggestion
    public int approveStudent(Long sessionId, Long id) {
        Student student = studentRepository.findById(id);
        student.approve();
        return studentRepository.save(student);
    }

    @Transactional
    public int approveCancelStudent(Long sessionId, Long id) {
        Student student = studentRepository.findById(id);
        student.cancal();
        return studentRepository.save(student);
    }
```

위와 같이 구현하는 것은 어떨까?
현재 구현한 버전과 위와 같이 구현했을 때의 장단점은??

Reply:
장점은 학생이라는 객체의 상태를 학생에게 맡기는 것으로 보입니다.
또한, 가독성 또한 좋아진 것 같습니다.

단점은 두번의 DB Connection이 아닐까요??